File: templates/mcp-server-ts/skeleton/src/transports/stdio.ts
import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js";
import { logger } from "../logger.js";

/**
 * Start the MCP server using stdio transport.
 *
 * The MCP client spawns this process as a subprocess and communicates
 * over stdin/stdout. This is the standard transport for desktop
 * integrations (Claude Desktop, Cursor, etc.).
 */
export async function start(server: McpServer): Promise<void> {
  const transport = new StdioServerTransport();
  await server.connect(transport);
  logger.info("server.transport", { transport: "stdio" });
}

File: templates/mcp-server-python/skeleton/src/transports/stdio.py
"""Stdio transport for the MCP server.

The MCP client spawns this process as a subprocess and communicates
over stdin/stdout. This is the standard transport for desktop
integrations (Claude Desktop, Cursor, etc.).
"""

from mcp.server.fastmcp import FastMCP

from ..logger import get_logger

logger = get_logger("transport.stdio")


def start(server: FastMCP) -> None:
    """Start the MCP server using stdio transport.

    Args:
        server: The configured FastMCP server instance.
    """
    logger.info("server.transport", transport="stdio")
    server.run(transport="stdio")
